concent_api/core/views.py

```diff
@@ -0,0 +1,223 @@
+import json
+import datetime
+
+from django.http                    import HttpResponse
+from django.views.decorators.http   import require_POST
+from django.utils                   import timezone
+from django.conf                    import settings
+
+from utils.api_view                 import api_view, Http400
+from .models                        import Message, MessageStatus
+
+
+@api_view
+@require_POST
+def send(request, message):
+    data = message
+    validate_message(data)
+
+    if data['type'] == "MessageForceReportComputedTask":
+        validate_message_task_to_compute(data['message_task_to_compute'])
```

Comment:
W metodach send i receive w views w tych ifach jest trochę powtarzającego się kodu. Można by
   a) zrefactorować to do Class Based View.
   b) wyciągnąć powtarzającą się część poza if-y
   c) to co jest w ifach wrzucić do osobnych funkcji/metod, wtedy kod samego send/receive które z tego co zgaduję jeszcze się skomplikuje w przyszłości upraszamy do:
		if message.is_type_message_force_report_computed_task():
			do_a()
		elif message.is_type_message_reject_report_computed_task():
			do_b()

Reply:
I agree, this needs to be refactored. I also don't like how spaghetti-like it's becoming.

I don't think these views are a good match for CBVs though. They're pretty simple and there's no matching CBV class for what they're meant to do. What we're doing here is not a typical view that serves HTML or a REST API view. I'd rather they stay as dead-simple functional views.

Extracting common logic into functions is a good idea though. We won't be amending this code now since @kbeker is already working on top of it and rewriting it but when he finishes with his current branch you could help him to refactor this code.

Reply:
:+1: